src/validator.ts

```diff
@@ -4,12 +4,32 @@ import {mapDeep} from './utils'
 type Rule = (schema: JSONSchema) => boolean | void
 const rules = new Map<string, Rule>()
 
+rules.set('Enum members and x-enum-varnames must be of the same length', schema => {
+  if (schema.enum && schema['x-enum-varnames'] && schema.enum.length !== schema['x-enum-varnames'].length) {
+    return false
+  }
+})
+
+/**
+ * `tsEnumNames` is being replaced by `x-enum-varnames`, but we'll leave this
+ * validator until `tsEnumNames` is removed.
+ */
 rules.set('Enum members and tsEnumNames must be of the same length', schema => {
   if (schema.enum && schema.tsEnumNames && schema.enum.length !== schema.tsEnumNames.length) {
     return false
   }
 })
 
+rules.set('x-enum-varnames must be an array of strings', schema => {
+  if (schema['x-enum-varnames'] && schema['x-enum-varnames'].some(_ => typeof _ !== 'string')) {
+    return false
+  }
+})
+
+/**
+ * `tsEnumNames` is being replaced by `x-enum-varnames`, but we'll leave this
+ * validator until `tsEnumNames` is removed.
+ */
```

Comment:
Please make sure this is still covered by tests.

Reply:
Should I copy `enumValidation.1.ts` and `enumValidation.2.ts`, but `sed "s/tsEnumNames/x-enum-varnames/"` in the new test files?

Reply:
Sounds ok to me, or merge into a single test file that covers all the cases (x-, ts-, x- and ts-, etc.).